catalog/app/containers/Bucket/UploadDialog.js

```diff
@@ -623,14 +623,19 @@ function WorkflowInput({ input, meta, workflowsConfig }) {
   )
 }
 
-function SchemaFetcher({ children, bucket, schemaUrl }) {
+function SchemaFetcher({ children, schemaUrl }) {
   const s3 = AWS.S3.use()
-  const data = useData(requests.metadataSchema, { s3, bucket, schemaUrl })
+  const data = useData(requests.metadataSchema, { s3, schemaUrl })
   const res = React.useMemo(
     () =>
       AsyncResult.mapCase(
         {
           Ok: (schema) => ({ schema, validate: validateMeta(schema) }),
+          Err: (responseError) => ({
+            responseError,
+            schema: null,
```

Comment:
this isnt required here i believe

Reply:
alternative approach: dont use `mapCase`, use something like this instead:
```javascript
  const res = React.useMemo(
    () =>
      data.case({
        Ok: (schema) => AsyncResult.Ok({ schema, validate: validateMeta(schema) }),
        Err: (responseError) => AsyncResult.Ok({ responseError, validate: validateMeta(null) }),
        _: R.identity,
      }),
    data.case,
  )
```
and then later
```javascript
<SchemaFetcher schemaUrl={R.pathOr('', ['schema', 'url'], values.workflow)}>
  {AsyncResult.case({
    Ok: (props) => (
      <RF.Field
        component={MetaInput}
        bucket={bucket}
        name="meta"
        validateFields={['meta']}
        isEqual={R.equals}
        {...props}
      />
    ),
    // no Err here bc Err is mapped to Ok inside SchemaFetcher
    _: () => ...
  })}
</SchemaFetcher>
```

Reply:
I tried to find out how to do the same with `mapCase`, thanks for example

Reply:
Second `data.case` in your exarmple is redundant, right?

Reply:
it's a `useMemo` dependency, should be an array)